dot/sync/disjoint_block_set.go

```diff
@@ -125,21 +172,17 @@ func (s *disjointBlockSet) addHeader(header *types.Header) error {
 	defer s.Unlock()
 
 	hash := header.Hash()
-	b, has := s.blocks[hash]
-	if has {
+	if b, has := s.blocks[hash]; has {
 		b.header = header
+		b.clearAt = time.Now().Add(ttl)
```

Comment:
Let's have a field in `disjointBlockSet` as `timeNow func() time.Time` and set it it its constructor. You can even inject `time.Now` to the constructor so even the constructor can be used in tests.

And use `b.clearAt = s.timeNow().Add(ttl)` here

Reply:
updated